rust/Cargo.toml

```diff
@@ -11,10 +11,24 @@ homepage = "https://roualdes.github.io/bridgestan/latest/"
 [dependencies]
 libloading = "0.8.0"
 thiserror = "1.0.40"
+ureq = { version = "2.7", optional = true }
+tar = { version = "0.4", optional = true }
+flate2 = { version = "1.0", optional = true }
+dirs = { version = "5.0", optional = true }
+path-absolutize = { version = "3.1", optional = true }
+log = { version = "0.4", optional = true }
+
+[features]
+compile-stan-model = ["dep:ureq", "dep:tar", "dep:flate2", "dep:dirs", "dep:path-absolutize", "dep:log"]
```

Comment:
Thanks for the changes. I'll let @aseyboldt comment on the ones he suggested, but this is looking pretty good from my end. 

I've given this some more thought and realized that almost all of these dependencies are not specific to _compilation_, but rather they're all for the "download the source if you don't have it" feature (right?).

Would it be reasonable to have _that_ be what is gated behind a feature? If the crate wasn't compiled with that and can't find the sources, it could unconditionally return the DownloadFailed error with a message saying so.

Reply:
this is a good point, also helps on the security matter mentioned by @aseyboldt